gecode/flatzinc/flatzinc.cpp

```diff
@@ -1999,6 +2009,182 @@ namespace Gecode { namespace FlatZinc {
 
   bool
   FlatZincSpace::slave(const MetaInfo& mi) {
+    if (mi.type() == MetaInfo::RESTART) {
+      if (restart_data().mark_complete) {
+        // Fail the space
+        this->fail();
+        // Return true to signal we are in the global search space
+        return true;
+      }
+
+      bool ret = false;
+      if (on_restart_iv.size() > 0) {
+        int base = 0;
+
+        // Assign the sol_int values (use lb if no solution is known)
+        const FlatZincSpace* last = static_cast<const FlatZincSpace*>(mi.last());
+        assert(last == nullptr || last->on_restart_iv.size() == restart_data().on_restart_iv_sol);
+        for (int i = 0; i < restart_data().on_restart_iv_sol; ++i) {
+          IntVar& outVar = on_restart_iv[base + restart_data().on_restart_iv_sol + i];
+          int solVal = (last != nullptr) ? last->on_restart_iv[i].val() : on_restart_iv[base + i].min();
+          rel(*this, outVar, IRT_EQ, solVal);
+        }
+        base += restart_data().on_restart_iv_sol * 2;
+
+        // Assign last_val_int values
+        for (size_t i = 0; i < restart_data().last_val_int.size(); ++i) {
+          rel(*this, on_restart_iv[base + i], IRT_EQ, restart_data().last_val_int[i]);
+        }
+        base += restart_data().last_val_int.size();
+
+        // Assign uniform_int random values 
+        for (size_t i = 0; i < restart_data().uniform_range_int.size(); ++i) {
+          const auto& range = restart_data().uniform_range_int[i];
+          const int rndVal = range.first + _random(static_cast<unsigned int>(range.second - range.first));
+          rel(*this, on_restart_iv[base + i], IRT_EQ, rndVal);
+        }
+        base += restart_data().uniform_range_int.size();
+
+        // Set the status value
+        if (restart_data().on_restart_status) {
+          assert(base == on_restart_iv.size() - 1);
+          IntVar& restart_status = on_restart_iv[base];
+          switch(mi.reason()) {
+            case MetaInfo::RR_INIT:
+              assert(!mi.last());
+              rel(*this, restart_status, IRT_EQ, 1); // 1: START
```

Comment:
These magic numbers could be defined as constants with the names START, SAT, and so on.

The comments are descriptive enough.